src/hooks/useAuthForm.js

```diff
@@ -0,0 +1,30 @@
+import { useReducer } from "react";
+import { authFormReducer } from "../reducers";
+
+export const useAuthForm = (initialForm) => {
+  const [authFormState, authFormDispatch] = useReducer(authFormReducer, {
+    loading: false,
+    showPassword: false,
+    credentials: initialForm,

```

Comment:
Suggestion assuming initialForm is related to user credentials you could use some more specific name maybe something sort of initial values or default.

Reply:
Okay, l will update that